### ISSUE TITLE
7 Days to Die - Serverinstall Debian 9

### ISSUE THREAD

**Issue by DerMoki** [ValveSoftware/steam-for-linux#6692]
#### Your system information

* up to date SteamCMD
* Debian 9 
* Opted into Steam client beta?: [Yes]
* Have you checked for system updates?: [Yes]

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen. Please link any large code pastes as a [Github Gist](https://gist.github.com/)

#### Steps for reproducing this issue:

Hey Steamsupport!

I like to install 7DaystoDie to my Debian Linux Server, i have a running server since years,but after update 17 in may i cannot install the Server anymore.

I login to Steamcmd , (with a other account!) but there is a timeout:

Steam>login OOTServer
Logging in user 'OOTServer' to Steam Public ...
Logged in OK
Waiting for user info...FAILED. Timed out.
OK

Whats can i do, this is not a Game-Error, its from Steamcmd

**Comment by Claiyc** [comment]
Same issue here, using anonymous login.

connection_log.txt:
IPv6 HTTP connectivity test (ipv6check-http.steamcontent.com / (...) - FAILURE (Invalid)
IPv6 UDP connectivity test (ipv6check-udp.steamcontent.com / (...) - TIMEOUT

(Not using ipv6 at all / no plans on using ipv6)

appinfo_log.txt:
[2019-12-01 18:01:02] RequestAppInfoUpdate: nNumAppIDs 1
[2019-12-01 18:01:02] Change number 0->7299591, apps: 0/0, packages: 0/0
[2019-12-01 18:01:02] Requesting 1 apps, 1 packages (meta data, 1024 prev attempts)
[2019-12-01 18:01:03] Requesting 1 apps, 1 packages (full data, 1024 prev attempts)
[2019-12-01 18:01:03] UpdatesJob: finished OK, nAppsUpdated 1 (3 KB), nPackagesUpdated 1 (60 KB)
[2019-12-01 18:01:03] Requesting 1034 apps, 0 packages (meta data, 1024 prev attempts)
[2019-12-01 18:01:04] Requested 551 access tokens, 38 received, 513 denied
[2019-12-01 18:01:05] Updated SHAs with 38 new access tokens
[2019-12-01 18:01:05] Downloaded 46 apps via HTTP, 1049 KB (243 KB compressed)
[2019-12-01 18:01:05] Requesting 988 apps, 0 packages (full data, 1024 prev attempts)
[2019-12-01 18:01:18] Starting async write /appcache/appinfo.vdf (791011 bytes)
[2019-12-01 18:01:32] RequestAppInfoUpdate: nNumAppIDs 1
[2019-12-01 18:02:02] UpdatesJob: finished Failure, nAppsUpdated 46 (1049 KB), nPackagesUpdated 0 (0 KB)
[2019-12-01 18:02:03] Starting async write /appcache/appinfo.vdf (791011 bytes)

Thats all useful information I can get from the logs.

Deleting appinfo.vdf & packageinfo.vdf from /appcache/ before every single steamcmd login helped in the past, but not anymore.

**Comment by UponABlackStar** [comment]
Got the same issue for a CS:GO server:

`[  0%] Checking for available updates...
[----] Verifying installation...
Steam Console Client (c) Valve Corporation
-- type 'quit' to exit --
Loading Steam API...OK.

Connecting anonymously to Steam Public...Logged in OK
Waiting for user info...FAILED. Timed out.
OK
CWorkThreadPool::~CWorkThreadPool: work processing queue not empty: 1 items discarded.`

I had the issue before, but back then I just removed the "Steam" and the ".steam" folder in the home directory before updating the game, but right now this does not seem to fix it.

**Comment by struppinet** [comment]
Same here. Using this docker image: didstopia/7dtd-server

Last Update it helped to delete the .steam and Steam folders. Now it even fails on the waiting for user info...

**Comment by struppinet** [comment]
Tried the official steamcmd docker container: same result ...
docker run -it --name=steamcmd cm2network/steamcmd bash

**Comment by Claiyc** [comment]
An update to Debian 10 (from Debian 9) fixed the issue for me.

**Comment by UponABlackStar** [comment]
> 
> 
> An update to Debian 10 (from Debian 9) fixed the issue for me.

Sadly thats not really an option for me, because my server is already running on Buster.

**Comment by Xperthobbit** [comment]
Using: CentOS Linux release 8.0.1905 (Core).
After updating/upgrading still cannot login anonymous.

`[  0%] Checking for available updates...
[----] Verifying installation...
Steam Console Client (c) Valve Corporation
-- type 'quit' to exit --
Loading Steam API...OK.

Connecting anonymously to Steam Public...Logged in OK
Waiting for user info...FAILED. Timed out.
OK`

> I had the issue before, but back then I just removed the "Steam" and the ".steam" folder in the home directory before updating the game, but right now that does not seem to fix it.

This used to be my solution also but now it doesn't seem to work.

**Comment by bviktor** [comment]
Can confirm. Works just fine on CentOS 7 but consistently fails on CentOS 8.

For a while, the install itself worked, and all consecutive updates failed.

For a day or two, I haven't even been able to even install anything.

**Comment by bviktor** [comment]
Tried to disable SELinux, disable firewalld, run as root, install a bunch of packages, add a lot of folders to LD_LIBRARY_PATH, nope, nothing helps.

**Comment by bviktor** [comment]
https://github.com/ValveSoftware/steam-for-linux/issues/6326#issuecomment-506124600

So the dudes apparently think those are the only kernels in existence. C8 runs on

```
$ uname -a
Linux c8test 4.18.0-80.11.2.el8_0.x86_64 #1 SMP Tue Sep 24 11:32:19 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

And yes, those counters do increase:

```
$ sudo nstat -az | grep -i wqueue
TcpExtTCPWqueueTooBig           43                 0.0
```

So maybe #6326 isn't fixed after all.

**Comment by bviktor** [comment]
OMG what a mess. I actually managed to solve it but it's ridiculous. Details here:

https://wiki.centos.org/Manuals/ReleaseNotes/CentOSStream

TLDR due to CentOS 8 bug kernels weren't receiving updates. That's why this bug in the kernel was left unpatched. Solution:

```
sudo dnf install -y centos-release-stream
sudo dnf -y update
sudo reboot
```

Then it's quite a bump:

```
$ uname -a
Linux c8test 4.18.0-147.6.el8.x86_64 #1 SMP Tue Oct 15 15:19:32 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

Aaand that's it, steamcmd immediately starts to work.

**Comment by vega0** [comment]
Thank you brotha

**Comment by ravenx99** [comment]
Can confirm, on Debian 9, an update last night and a reboot this morning did **not** fix.
Upgrading kernel to linux-image-4.9.0-11-amd64, which doesn't seem to have been available last night, **did fix**.

**Comment by UponABlackStar** [comment]
Is there a similar solution for Debian systems?

**Comment by ravenx99** [comment]
I just did an `apt-get update && apt-get dist-upgrade` to upgrade the kernel to the latest on Debian 9.

**Comment by UponABlackStar** [comment]
On my Buster installation the problem still exists:
`uname -a`
`Linux v22019063595491105 4.19.0-5-amd64 #1 SMP Debian 4.19.37-5 (2019-06-19) x86_64 GNU/Linux`

**Comment by DerMoki** [comment]
Update to Debian 10, but the problem still exists too

**Comment by ertug2** [comment]
Issue has been happening at random on many of my nodes regardless of location, region or routing for the past 2-3 months. Debian here as well. 

Everything is operating behind our ISP and there aren't any issues between our networks.

A temporary fix was to login to a persistent steam account, log out then log into anonymous (some games force that you are logged into anonymous to download content). On some isolated nodes, that isn't working any longer as of two to three days ago. 

Seems to have come right after the new debug information printed in SteamCMD.

```
>
Steam>login anonymous

Connecting anonymously to Steam Public...Loaded client id: 18389946503649214524
Listening for IPv4 broadcast on: 27036
Logged in OK
Waiting for user info...FAILED. Timed out.
OK
```

Anybody aware of any cache location for steamworks/steamcmd that I can try clearing?

**Comment by VergilPrime** [comment]
This is in fact not OK.

**Comment by ertug2** [comment]
Agreed.

**Comment by danielgran** [comment]
Same here

**Comment by nvtkaszpir** [comment]
got the same issue with two different accounts on the same linux host:
- mine works, but always asks for password and steamguard code, I never managed to save my password on the host:
```text
Logging in user 'nvt_k...' to Steam Public...FAILED (invalid cached credentials)

password: OK
Waiting for client config...OK
Waiting for user info...OK
app_update 223250  validate
 Update state (0x0) unknown, progress: 0.00 (0 / 0)
Error! App '223250' state is 0x204 after update job.
app_update 1290 verify
Success! App '1290' already up to date.
```

- account of my friend at first logs (asks for steamguard) and after login it never prompts for password, but it responds very slowly and then shows timeouts when retreiving user data:
```

Logging in user 'bart..' to Steam Public...OK
Waiting for client config...ERROR! Timed out waiting for AppInfo update.
OK
Waiting for user info...FAILED (Timed out)
OK
app_update 223250  validate
ERROR! Timed out waiting for AppInfo update.
```

Nothing interesting in the logs, maybe I should enable debug/trace logs (how?)

**Comment by Vennnot** [comment]
Hey everyone! I've been having a similar issue. I'm not really tech savvy and I've been struggling on ubuntu 24.04 LTS.

I tried reinstalling, updating packages and everything else ChatGPT and the internet recommends. No matter what I do though, the error is always the same. The weird thing is, on the first time I ever ran it to test the console worked normally. After I closed it and ever since then it's been a nightmare.

From a fresh install this is what happens:

```
./steamcmd
Redirecting stderr to '/home/daniel/Steam/logs/stderr.txt'
Looks like steam didn't shutdown cleanly, scheduling immediate update check
ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
[  0%] Checking for available update...
[----] Downloading update (0 of 69,768 KB)...
[  0%] Downloading update (0 of 69,768 KB)...
[  0%] Downloading update (0 of 69,768 KB)...
[  0%] Downloading update (2,776 of 69,768 KB)...
[  3%] Downloading update (6,008 of 69,768 KB)...
[  8%] Downloading update (9,380 of 69,768 KB)...
[ 13%] Downloading update (12,424 of 69,768 KB)...
[ 17%] Downloading update (15,644 of 69,768 KB)...
[ 22%] Downloading update (19,203 of 69,768 KB)...
[ 27%] Downloading update (22,436 of 69,768 KB)...
[ 32%] Downloading update (25,666 of 69,768 KB)...
[ 36%] Downloading update (29,215 of 69,768 KB)...
[ 41%] Downloading update (32,470 of 69,768 KB)...
[ 46%] Downloading update (35,678 of 69,768 KB)...
[ 51%] Downloading update (39,218 of 69,768 KB)...
[ 56%] Downloading update (42,451 of 69,768 KB)...
[ 60%] Downloading update (45,680 of 69,768 KB)...
[ 65%] Downloading update (49,189 of 69,768 KB)...
[ 70%] Downloading update (52,404 of 69,768 KB)...
[ 75%] Downloading update (55,642 of 69,768 KB)...
[ 79%] Downloading update (59,183 of 69,768 KB)...
[ 84%] Downloading update (61,975 of 69,768 KB)...
[ 88%] Downloading update (63,623 of 69,768 KB)...
[ 91%] Downloading update (65,451 of 69,768 KB)...
[ 93%] Downloading update (67,123 of 69,768 KB)...
[ 96%] Downloading update (68,783 of 69,768 KB)...
[ 98%] Downloading update (69,768 of 69,768 KB)...
[100%] Download Complete.
[----] Applying update...
[----] Extracting package...
[----] Extracting package...
[----] Extracting package...
[----] Extracting package...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Cleaning up...
[----] Update complete, launching...
```

then...

```
Redirecting stderr to '/home/daniel/Steam/logs/stderr.txt'
Logging directory: '/home/daniel/Steam/logs'
[  0%] Checking for available updates...
[----] Verifying installation...
[  0%] Downloading update...
[  0%] Checking for available updates...
[----] Downloading update (0 of 48,921 KB)...
[  0%] Downloading update (1,464 of 48,921 KB)...
[  2%] Downloading update (4,566 of 48,921 KB)...
[  9%] Downloading update (8,026 of 48,921 KB)...
[ 16%] Downloading update (10,910 of 48,921 KB)...
[ 22%] Downloading update (13,786 of 48,921 KB)...
[ 28%] Downloading update (16,328 of 48,921 KB)...
[ 33%] Downloading update (18,884 of 48,921 KB)...
[ 38%] Downloading update (21,917 of 48,921 KB)...
[ 44%] Downloading update (24,488 of 48,921 KB)...
[ 50%] Downloading update (27,088 of 48,921 KB)...
[ 55%] Downloading update (29,912 of 48,921 KB)...
[ 61%] Downloading update (32,510 of 48,921 KB)...
[ 66%] Downloading update (35,109 of 48,921 KB)...
[ 71%] Downloading update (37,654 of 48,921 KB)...
[ 76%] Downloading update (39,528 of 48,921 KB)...
[ 80%] Downloading update (41,136 of 48,921 KB)...
[ 84%] Downloading update (42,869 of 48,921 KB)...
[ 87%] Downloading update (44,438 of 48,921 KB)...
[ 90%] Downloading update (46,013 of 48,921 KB)...
[ 94%] Downloading update (47,752 of 48,921 KB)...
[ 97%] Downloading update (48,921 of 48,921 KB)...
[100%] Download complete.
[----] Extracting package...
[----] Extracting package...
[----] Extracting package...
[----] Extracting package...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Installing update...
[----] Cleaning up...
[----] Update complete, launching Steamcmd...
UpdateUI: skip show logo
```

and then...


```
daniel@danielsdesktop:~/Downloads/steamworks_sdk_159/sdk/tools/ContentBuilder/builder_linux/linux32$ ./steamcmd
Redirecting stderr to '/home/daniel/Steam/logs/stderr.txt'
Logging directory: '/home/daniel/Steam/logs'
[  0%] Checking for available updates...
[----] Verifying installation...
UpdateUI: skip show logodlmopen steamconsole.so failed: steamconsole.so: cannot open shared object file: No such file or directory
[----] !!! Fatal Error: Failed to load steamconsole.so
```

Then it's always the same error. Checking on the logs the only message is this one:

`Failed to load steamconsole.so - dlerror(): ./steamcmd: undefined symbol: WORKAROUND_GdbSymbolLoadHook`

I've tried all the classic things including:

- updating 32 bit libraries
- reinstalling them
- reinstalling steamcmd
- reinstalling steam
- purging everything and THEN reinstalling.

I'm at the end of my rope. I don't want to buy windows and move all my files just to upload to steam. My friend gave the this Ubuntu PC. Any help is appreciated. Thanks!

**Comment by GrimLothar** [comment]
i know noone will see this... but this started happening on macOS sometime within the last week (i just noticed it today, last login was last week)

**Comment by AdamLearns** [comment]
> this started happening on macOS sometime within the last week

@GrimLothar I hit this same issue, and simply closing Steam seems to have fixed it for me. Barring that, it's possible (although unconfirmed) that it was my DNS-level ad-blocker.

**Comment by GrimLothar** [comment]
Got fixed with the latest update on steamcmd. Feel free to close!

**Comment by ohsqueezy** [comment]
I was able to fix the `Waiting for user info...FAILED. Timed out.` failure by deleting my `userdata/` folder. For me the folder is located at `~/.steam/debian-installation/userdata`. I backed up the folder just in case.
